pandas-stubs/core/frame.pyi

```diff
@@ -1711,7 +1711,7 @@ class DataFrame(NDFrame, OpsMixin):
     ) -> DataFrame: ...
     def filter(
         self,
-        items: list | None = ...,
+        items: Iterable | None = ...,
```

Comment:
`Iterable` is too wide.   That will accept a plain string.  Can you use `ListLike` here

While the pandas code accepts a set, that's not the intent and why the docs say "list-like".  That would also mean you should remove the test that passes a set.

Reply:
I used `ListLike` in the first iteration but then mypy fails for `set` or `generator`

```console
tests\test_frame.py:323: error: Value of type variable "ListLike" of "filter" of "DataFrame" cannot be "set[str]"  [type-var]
tests\test_frame.py:325: error: Value of type variable "ListLike" of "filter" of "DataFrame" cannot be "Generator[str, None, None]"  [type-var]
```

I skimmed the code for something in the middle... Maybe `Iterable[str]`?

Reply:
Since `items` is converted into an `Index` it's very flexible. And `Index.__new__` has a very complicated signature with multiple overloads.
Your call as to which way to go; stricter and have false negatives, or loose and have false positives.

Reply:
the rule we are using in the stubs is to base things on the docs.  So while `filter` does accept a `set`, it probably shouldn't, given the docs.  So I am for removing the test for sets, and if you want to include a generator, you can use `ListLike | Generator[Hashable, None, None]`

Reply:
Done